ingress/controllers/nginx/nginx/template/template.go

```diff
@@ -54,48 +56,86 @@ var (
 	}
 )
 
-func (ngx *Manager) loadTemplate() error {
-	tmpl, err := template.New("nginx.tmpl").Funcs(funcMap).ParseFiles(tmplPath)
+// Template ...
+type Template struct {
+	tmpl    *text_template.Template
+	watcher *fsnotify.Watcher
+}
+
+//NewTemplate returns a new Template instance or an
+//error if the specified template file contains errors
+func NewTemplate(file string) (*Template, error) {
+	tmpl, err := text_template.New("nginx.tmpl").Funcs(funcMap).ParseFiles(file)
+	if err != nil {
+		return nil, err
+	}
+
+	// we need to watch the template file for changes
+	watcher, err := fsnotify.NewWatcher()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	ngx.template = tmpl
-	return nil
+
+	err = watcher.Add(file)
+	if err != nil {
+		return nil, err
+	}
+
+	template := &Template{
+		tmpl:    tmpl,
+		watcher: watcher,
+	}
+
+	go func() {
```

Comment:
I would vastly prefer if we did this in a compsable way. Perhaps we can have a sidecar that watches the config map, notices a change, and forces a reload on the nginx controller through a /reload http endpoint ? that would make for a nice reusable component. 
```
$ watch-configmaps --reload-url="url-in-same-netns" --file="/path/to/file"
```

When we upgrade to docker1.12 we will get shared pid namespaces and can consider replacing the --reload-url arg with just `pkill SIGHIUP nginx`. 

The *most* reusable way to do this right now would be to avoid even the http server, and do:
```
$ watch-configmaps --pod-name="foopod, defaults to self" \
    --container-name="nginx, defaults to first peer container in list of own pod" \
    --file="/path/to/file" \
    --script="/reload.sh"
```

That would kubectl exec in the give pod and exec the the signal to the given container/process. If you're interested you can download the kubectl client from: 
`wget https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl`

Or exec through a REST client like we do in our e2es: 
https://github.com/kubernetes/kubernetes/blob/4bb5fdc47f4c409f7d290b3af348fb135d0daf0f/test/e2e/framework/util.go#L3756

Quite a few people have asked for something like the last one. I would put this under ingress/ so it doesn't get lost when we split ingress out of contrib.